### ISSUE TITLE
Setting to disable autocomplete popups

### ISSUE THREAD

**Issue by readmemyrights** [supercollider/supercollider#7527]
## Motivation

When using supercollider with a screen reader, the autocomplete popups make it switch focus to an inaccessible list and one has to do something like switch to the post window and back to the editor to continue typing. I also in general find autocompletion distracting, so I would prefer if there was a way to turn them off, and perhaps have a shortcut to manually complete the current context.

## Description of Proposed Feature

An item in the preferences of supercollider IDE to disable autocompletion popups.

## Plan for Implementation

I am not intimately familiar with the internals of supercollider, but I doubt this would be hard to implement.

I am curious if there was any prior attempts to improve accessibility of the IDE, it works great with screen readers for the most part but some parts noteably the help browser and autocompletions are nearly completely inaccessible. I am not familiar with programming in QT unfortunately but I would gladly test and give suggestions.

**Comment by capital-G** [comment]
I think we didn't had a focus on accessibility so far, but I'd really like to put more emphasis on this, so thanks for the feedback! But great to hear that there aren't too much issues otherwise with the IDE - we are working on a new version of the docs, so we hope to also account accessibility there.

Would it be beneficial for you to completely disable auto-complete or would there be a way where auto-complete would be supportive as well?

**Comment by readmemyrights** [comment]
> Would it be beneficial for you to completely disable auto-complete or would there be a way where auto-complete would be supportive as well?

There are ways to make autocompletion accessible, I heard of programs that implement it well but I don't know any off the top of my head, I don't like autocompletion in general but not necessarily because it's an accessibility issue.

A possible middle ground could be manual completion, typing usually does nothing but if you press tab for example it brings up the menu of classes / methods that begin with what you typed so far, so in the usual case your typing doesn't get interrupted, but one can still do say Imp<tab> to complete Impulse. The main issue still though is the fact that the completion menu currently doesn't read anything when one moves through it with the arrow keys, and it doesn't switch the focus back when an option is selected.

I'll make another issue in regards to the help system some time later, hopefully not much more than a day or two. There's a strange regression, I want to say it was 3.13 to 3.14 but I am not quite sure which version it got introduced in, but the help system became much less useable within a single update. I'll explain it more in that issue though, after I do some testing.
